spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkUtil.java

```diff
@@ -292,4 +292,8 @@ public static List<Expression> partitionMapToExpression(
   public static String toColumnName(NamedReference ref) {
     return DOT.join(ref.fieldNames());
   }
+
+  public static boolean caseSensitive(SparkSession spark) {
+    return Boolean.parseBoolean(spark.conf().get("spark.sql.caseSensitive"));
```

Comment:
We don't need to take it in this PR, but I see hardcoded "spark.sql.caseSensitive" in a few places in the spark code, is there a good place to define this constant?  I know it's a general spark sql option and not Iceberg specific so `SparkSQLProperties` is probably not the right place

Reply:
I tried to partially address that by introducing this utility call and migrating all other places to use it.

Reply:
Ah I see I missed that this is public, then that should be good!